drivers/linstorvhdutil.py

```diff
@@ -120,14 +110,45 @@ def wrapper(*args, **kwargs):
             remote_args = {str(key): str(value) for key, value in remote_args.items()}
 
             try:
-                def remote_call():
-                    host_ref = self._get_readonly_host(vdi_uuid, device_path, node_names)
-                    return call_remote_method(self._session, host_ref, remote_method, device_path, remote_args)
-                response = util.retry(remote_call, 5, 2)
-            except Exception as remote_e:
-                self._raise_openers_exception(device_path, local_e or remote_e)
+                host_ref_attached = list(util.get_hosts_attached_on(self._session, [vdi_uuid]))[0]
+                if host_ref_attached:
+                    response = call_remote_method(
+                        self._session, host_ref_attached, remote_method, device_path, remote_args
+                    )
+                    log_successful_call('attached node', device_path, vdi_uuid, remote_method, response)
+                    return response_parser(self, vdi_uuid, response)
+            except Exception as e:
+                log_failed_call('attached node', 'master', device_path, vdi_uuid, remote_method, e)
+
+            try:
+                master_ref = list(self._session.xenapi.pool.get_all_records().values())[0]['master']
```

Comment:
We could use `util.get_master_ref()` directly here.
```suggestion
                master_ref = util.get_master_ref(self._session)
```

Reply:
fixed in [07ad622](https://github.com/xcp-ng/sm/pull/93/commits/07ad62291c4914692a28517c3713028f08b5d20b)